File: go-client-sdk/pkg/models/operations/getinvoicesupcoming.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/v2/pkg/models/shared"
)

type GetInvoicesUpcomingQueryParams struct {
	SubscriptionID string `queryParam:"style=form,explode=true,name=subscription_id"`
}

type GetInvoicesUpcomingRequest struct {
	QueryParams GetInvoicesUpcomingQueryParams
}

type GetInvoicesUpcomingResponse struct {
	ContentType              string
	StatusCode               int
	UpcomingPercent20Invoice *shared.UpcomingPercent20Invoice
}

File: typescript-client-sdk/src/sdk/models/operations/getinvoicesupcoming.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";
import { Type } from "class-transformer";


export class GetInvoicesUpcomingQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=subscription_id" })
  subscriptionId: string;
}

export class GetInvoicesUpcomingRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetInvoicesUpcomingQueryParams;
}

export class GetInvoicesUpcomingResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  upcomingPercent20Invoice?: shared.UpcomingPercent20Invoice;
}